internal/pkg/addon/addons.go

```diff
@@ -46,18 +46,25 @@ type Addons struct {
 
 	cachedTemplate    string
 	cachedTemplateErr error
+
+	bucket   string
+	uploader uploader
+
+	UploadAssets bool
 }
 
 // New creates an Addons object given a workload name.
-func New(wlName string) (*Addons, error) {
+func New(wlName string, bucket string, uploader uploader) (*Addons, error) {
```

Comment:
What do you think of creating two constructors?

```suggestion
// New creates a raw addons config without any modifications to the template.
func New(workload string) (*Addons, error) {

// NewPackager returns an addons config that can upload and swap local assets to s3.
func NewPackager(workload, bucket string, s3 Uploader) (*Addons, error) {
```

Reply:
great idea! added 😊